src/Cornelis/Vim.hs

```diff
@@ -23,115 +23,89 @@ vimFirstLine = 0
-  pure $ Pos line col'
+  unvimify b (Pos line col)
 
 -- | TODO(sandy): POSSIBLE BUG HERE. MAKE SURE YOU SET THE CURRENT WINDOW
 -- BEFORE CALLING THIS FUNCTION
-getpos :: Buffer -> Char -> Neovim env Pos
+getpos :: Buffer -> Char -> Neovim env AgdaPos
 getpos b mark = do
-  ObjectArray [_, objectToInt -> Just row, objectToInt -> Just col, _]
+  -- getpos gives us a (1,1)-indexed position!
+  ObjectArray [_, objectToInt @Int -> Just (oneIndexed -> line), objectToInt @Int -> Just (oneIndexed -> col), _]
     <- vim_call_function "getpos" $ V.fromList [ObjectString $ encodeUtf8 $ T.singleton mark]
-  -- getpos gives us a 1-indexed line, but that is the same way that
-  -- lines are indexed.
-  let line = LineNumber row
-  col' <- unvimifyColumn b line col
-  -- but the columns are one indexed!
-  pure $ Pos line $ offsetDiff col' $ Offset 1
+  unvimify b (Pos (zeroIndex line) (zeroIndex col))
 
 data SearchMode = Forward | Backward
   deriving (Eq, Ord, Show)
 
--- Like @searchpos@ from vim, but specialized to find one of many patterns
-searchpos :: Buffer -> [Text] -> SearchMode -> Neovim env Pos
+searchpos :: Buffer -> [Text] -> SearchMode -> Neovim env AgdaPos
 searchpos b pats dir = do
-  ObjectArray [objectToInt -> Just row, objectToInt -> Just col]
+  -- unlike getpos, these columns are 0 indexed W T F
+  ObjectArray [objectToInt @Int -> Just (oneIndexed -> line), objectToInt @Int -> Just (zeroIndexed -> col)]
     <- vim_call_function "searchpos" $ V.fromList
         [ ObjectString $ encodeUtf8 $ T.intercalate "\\|" pats
         , ObjectString $ encodeUtf8 $ case dir of
             Forward -> "n"
             Backward -> "bn"
         ]
-  -- getpos gives us a 1-indexed line, but that is the same way that
-  -- lines are indexed.
-  let line = LineNumber row
-  col' <- unvimifyColumn b line col
-  -- unlike getpos, these columns are 0 indexed W T F
-  pure $ Pos line col'
-
+  unvimify b (Pos (zeroIndex line) col)
 
-setWindowCursor :: Window -> Pos -> Neovim env ()
+setWindowCursor :: Window -> AgdaPos -> Neovim env ()
 setWindowCursor w p = do
   b <- window_get_buffer w
-  Pos (LineNumber l) c <- vimifyPositionM b p
-  -- lines are 1-indexed, but window_set_cursor also wants a 1-index, so we're
-  -- cool to not call 'getVimLineNumber' here.
-  window_set_cursor w (fromIntegral l, c)
+  Pos l c <- vimify b p
+  window_set_cursor w (fromOneIndexed (oneIndex l), fromZeroIndexed c)
 
-replaceInterval :: Buffer -> Pos -> Pos -> Text -> Neovim env ()
-replaceInterval b start end str
+replaceInterval :: Buffer -> Interval AgdaPos -> Text -> Neovim env ()
+replaceInterval b ival str
   = do
-    (sl, sc) <- fmap positionToVim $ vimifyPositionM b start
-    (el, ec) <- fmap positionToVim $ vimifyPositionM b end
-    nvim_buf_set_text b sl sc el ec $ V.fromList $ T.lines str
-
+    Interval (Pos sl sc) (Pos el ec) <- traverse (vimify b) ival
+    nvim_buf_set_text b (from0 sl) (from0 sc) (from0 el) (from0 ec) $ V.fromList $ T.lines str
```

Comment:
I know this is no worse than the original function, but the indexing scares me here. Maybe in a separate PR we make explicit wrappers for each vim API function and give them types that correspond to how their indexing is supposed to work?